lib/rubocop/cop/rspec/mock_not_stub.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that message expectations are not combined with message stubs.
+      #
+      # @example
+      #
+      #   # bad
+      #   expect(foo).to receive(:bar).with(42).and_return("hello world")
+      #
+      #   # good (without spies)
+      #   allow(foo).to receive(:bar).with(42).and_return("hello world")
+      #   expect(foo).to receive(:bar).with(42)
+      #
+      class MockNotStub < Base
+        MSG = 'Do not stub your mock.'
+
+        def_node_matcher :message_expectation?, <<-PATTERN
+          {
+            (send nil? { :receive :receive_message_chain } ...)
+            (send (send nil? :receive ...) :with ...)
```

Comment:
not sure this would work in all possible chains, like `receive.once.with()`
We need a helper for catching message expectations, see ReceiveCounts as an example of cop dealing with chains

Reply:
It for sure won't. Those count and ordering constraints will add complexity that I don't really want to deal with right now. I fully acknowledge this cop is not perfect.